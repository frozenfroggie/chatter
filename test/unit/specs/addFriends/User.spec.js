import { shallowMount } from '@vue/test-utils'
import User from '@/components/addFriends/User.vue'

describe('User', () => {
  it('renders user name in div.conversationName when props.user passed', () => {
    const user = {
      username: 'Christianna Lynn',
    }
    const wrapper = shallowMount(User, {
      propsData: { user },
      stubs: ['Icon']
    })
    expect(wrapper.find('div.conversationName').text()).toBe(user.username)
  })
  it('matches snapshot', () => {
    const user = {
      name: 'Christianna Lynn',
      isOnline: true
    }
    const wrapper = shallowMount(User, {
      propsData: { user },
      stubs: ['Icon']
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
