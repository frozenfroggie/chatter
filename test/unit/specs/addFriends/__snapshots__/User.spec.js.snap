// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches snapshot 1`] = `
<li class="conversationLabel">
    <div class="avatarContainer">
        <!---->
    </div>
    <div class="content">
        <div class="conversationName">

        </div>
    </div>
    <!---->
</li>
`;
